packages/cli/src/databases/migrations/common/1717498465932-RefactorExecutionIndices.ts

```diff
@@ -0,0 +1,87 @@
+import type { MigrationContext, ReversibleMigration } from '@/databases/types';
+
+/**
+ * Add new indices:
+ *
+ * - `status, startedAt` for default query at `ExecutionRepository.findManyByRangeQuery`
+ * - `workflowId, status, startedAt` for filtered query at `ExecutionRepository.findManyByRangeQuery`
+ * - `waitTill, status` for regular queries at `ExecutionRepository.getWaitingExecutions`
+ *
+ * Remove unused indices in sqlite:
+ *
+ * - `stoppedAt` (duplicate with different casing)
+ * - `waitTill`
+ * - `status, workflowId`
+ *
+ * Remove unused indices in MySql:
+ *
+ * - `status`
+ *
+ * Remove unused indices in all DBs:
+ *
+ * - `stopped_at`
+ * - `waitTill, id`
+ * - `workflowId, id`
+ *
+ * Keep index as is:
+ *
+ * - `deletedAt` for query at `ExecutionRepository.hardDeleteSoftDeletedExecutions`
+ */
+export class RefactorExecutionIndices1717498465932 implements ReversibleMigration {
+	async up({ schemaBuilder, isPostgres, isSqlite, isMysql }: MigrationContext) {
+		await schemaBuilder.createIndex('execution_entity', ['status', 'startedAt']);
+		await schemaBuilder.createIndex('execution_entity', ['workflowId', 'status', 'startedAt']);
+		await schemaBuilder.createIndex('execution_entity', ['waitTill', 'status']);
+
+		if (isSqlite) {
+			await schemaBuilder.dropIndex('execution_entity', ['stoppedAt'], {
+				customIndexName: 'idx_execution_entity_stopped_at',
+				skipIfMissing: true,
+			});
+
+			await schemaBuilder.dropIndex('execution_entity', ['waitTill'], {
+				customIndexName: 'idx_execution_entity_wait_till',
+				skipIfMissing: true,
+			});
+
+			await schemaBuilder.dropIndex('execution_entity', ['status', 'workflowId'], {
+				customIndexName: 'IDX_8b6f3f9ae234f137d707b98f3bf43584',
+				skipIfMissing: true,
+			});
+		}
+
+		if (isMysql) {
+			await schemaBuilder.dropIndex('execution_entity', ['status'], {
+				customIndexName: 'IDX_8b6f3f9ae234f137d707b98f3bf43584',
+				skipIfMissing: true,
+			});
+		}
+
+		// all DBs
+
+		await schemaBuilder.dropIndex('execution_entity', ['stoppedAt']);
```

Comment:
could there be a scenario where we might need to filter by `stoppedAt` instead? I get that we forgot to add the index for `startedAt`, but there must have been a reason why we added one for `stoppedAt`. don't we use this in pruning?

Reply:
Replied [here](https://n8nio.slack.com/archives/C0484MWCNN7/p1720684890546249?thread_ts=1719991363.012299&cid=C0484MWCNN7).